src/nplinker/genomics/antismash/antismash_downloader.py

```diff
@@ -0,0 +1,87 @@
+import os
+from os import PathLike
+from pathlib import Path
+import shutil
+from nplinker.logconfig import LogConfig
+from nplinker.utils import download_and_extract_archive
+from nplinker.utils import list_dirs
+from nplinker.utils import list_files
+
+logger = LogConfig.getLogger(__name__)
+
+# urls to be given to download antismash data
+ANTISMASH_DB_PAGE_URL = 'https://antismash-db.secondarymetabolites.org/output/{}/'
+ANTISMASH_DB_DOWNLOAD_URL = 'https://antismash-db.secondarymetabolites.org/output/{}/{}'
+
+# The antiSMASH DBV2 is for the availability of the old version, better to keep it.
+ANTISMASH_DBV2_PAGE_URL = 'https://antismash-dbv2.secondarymetabolites.org/output/{}/'
+ANTISMASH_DBV2_DOWNLOAD_URL = 'https://antismash-dbv2.secondarymetabolites.org/output/{}/{}'
+
+
+def download_and_extract_antismash_data(antismash_id: str,
+                                        download_root: str | PathLike,
+                                        extract_root: str | PathLike):
+    """Download and extract antiSMASH BGC archive for a specified genome.
+    The antiSMASH database (https://antismash-db.secondarymetabolites.org/)
+    is used to download the BGC archive. And antiSMASH use RefSeq assembly id
+    of a genome as the id of the archive.
+
+    Args:
+        antismash_id(str): The id used to download BGC archive from antiSMASH database.
+            If the id is versioned (e.g., "GCF_004339725.1") please be sure to 
+            specify the version as well. 
+        download_root(str | PathLike): Path to the directory to place downloaded archive in.
+        extract_root(str | PathLike): Path to the directory data files will be extracted to.
+            Note that an `antismash` directory will be created in the specified `extract_root` if
+            it doesn't exist. The files will be extracted to `<extract_root>/antismash/<antismash_id>` directory.
+
+    Raises:
+        ValueError: if download_root and extract_root dirs are the same.
+        ValueError: if <extract_root>/antismash/<refseq_assembly_id> dir is not empty.
+
+    Examples:
+        >>> download_and_extract_antismash_metadata("GCF_004339725.1", "/data/download", "/data/extracted")
+    """
+    download_root: Path = Path(download_root)
+    extract_root: Path = Path(extract_root)
```

Comment:
```suggestion
    download_root = Path(download_root)
    extract_root = Path(extract_root)
```
Not necessary to add type hit here.